plans/add_compiler.pp

```diff
@@ -0,0 +1,68 @@
+# @summary Add a new compiler to a PE architecture or replace an existing one with new configuration.
+# @param avail_group_letter _ Either A or B; whichever of the two letter designations the compiler is being assigned to
+# @param compiler_fqdn _ The FQDN and certname of the new compiler
+# @param dns_alt_names _ A comma_separated list of DNS alt names for the compiler
+# @param primary_server_fqdn _ The FQDN and certname of the primary Puppet server
+# @param postgresql_server_fqdn _ The FQDN and certname of the PE-PostgreSQL server with availability group $avail_group_letter
+plan peadm::add_compiler(
+  String[1] $avail_group_letter,
+  Peadm::SingleTargetSpec $compiler_fqdn,
```

Comment:
I'm tempted to say, maybe we should make this plan `peadm::add_compilers`, and support a `$compiler_hosts` array parameter instead of `$compiler_host`. This would make the plan more flexible for end users.

We could potentially achieve that by making a higher-level plan later though, so perhaps we finish the plan as-is, targeting deploying one compiler, and decide what to do next after it's done.

Reply:
Ok, let's keep it simple for now.